src/components/PostPageDetail/elements/Content.js

```diff
@@ -84,12 +85,26 @@ const Content = ({
     }
   }, [containerHeight, textHeight]);
 
+  const randomIndex = React.useMemo(() => {
+    return Math.floor(Math.random() * listFeedColor.length);
+  }, []);
+
   const isShortText = () => {
     return images_url.length <= 0 && item.post_type === POST_TYPE_STANDARD && !haveSeeMore;
   };
   const handleContainerPdp = () => {
     if (isShortText()) {
-      return styles.shortText(parentData?.bg);
+      const backgroundData = (() => {
+        if (parentData?.bg) {
+          return parentData.bg;
+        }
+        if (item.anon_user_info_color_code) {
+          return hexToRgb(item.anon_user_info_color_code, 0.25);
+        }
+        const newColor = listFeedColor[randomIndex];
+        return hexToRgb(newColor.bg, 0.25);
+      })();
+      return styles.shortText(backgroundData);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes in the `handleContainerPdp` function seem to be logically sound. It now checks for multiple conditions to determine the background color. However, there's a potential issue with the use of `randomIndex`. If the `listFeedColor` array length changes over the lifetime of the component, the `randomIndex` could potentially point to an out-of-bounds index. To avoid this, you could move the `randomIndex` calculation inside the `handleContainerPdp` function or ensure that the `listFeedColor` array length doesn't change during the component's lifetime.
```diff
+ const handleContainerPdp = () => {
+   if (isShortText()) {
+     const backgroundData = (() => {
+       if (parentData?.bg) {
+         return parentData.bg;
+       }
+       if (item.anon_user_info_color_code) {
+         return hexToRgb(item.anon_user_info_color_code, 0.25);
+       }
+       const newColor = listFeedColor[randomIndex];
+       return hexToRgb(newColor.bg, 0.25);
+     })();
+     return styles.shortText(backgroundData);
+   }
+   return {};
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->